tests/testthat/test-tbl-df.R

```diff
@@ -36,6 +36,22 @@ test_that("[ with 0 cols creates correct row names (#656)", {
   expect_identical(zero_row, as_tibble(iris)[0])
 })
 
+test_that("[ with 0 cols returns correct number of rows", {
+  iris_tbl <- as_tibble(iris)
+  nrow_iris <- nrow(iris_tbl)
+
+  expect_is(iris_tbl, "tbl_df")
```

Comment:
This test isn't about testing the behavior of `as_tibble()`, so I'd rather keep only the relevant parts.